.github/actions/msbuild/action.yml

```diff
@@ -0,0 +1,73 @@
+name: MSBuild .NET Framework Build with Static Code Analysis
+description: Builds the .NET Framework solution in the given directory with static code analysis.
+
+inputs:
+  directory:
+    required: false
+    default: .
+    description: Path to the directory where a solution file can be found.
+  verbosity:
+    required: false
+    default: quiet
+    description: Verbosity parameter for msbuild.
+  enable-code-analysis:
+    required: false
+    default: "true"
+    description: If set to "true", static code analysis is enabled during the build.
+  publish-version:
+    required: false
+    default: "1.${{ github.run_number }}.${{ github.run_attempt }}-${{ github.run_id }}"
+    description: The value given to `msbuild`'s `-Version` switch.
+  dotnet-build-switches:
+    required: false
+    default: ""
+    description: >
+      Additional command line switches given to `msbuild`. You must split these into multiple lines, even the
+      parameter value.
+  solution-or-project-path:
+    required: false
+    default: "*.sln"
+    description: >
+      The path of the solution or project file to be built. If you have exactly one .sln file in the current directory
+      then this can be omitted. Otherwise make sure to specify it to avoid an "MSB1008: Only one project can be
+      specified." error that is thrown when the `*.sln` wildcard results in multiple files. To build a project file you
+      must specify a path that ends with `.csproj`. An example: "./src/*Web/*.Web.csproj". The value is given to
+      PowerShell's `Get-ChildItem` cmdlet as-is, so grepping is still possible but a name with spaces must be escaped
+      separately.
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Scripts
+      shell: pwsh
+      run: |
+        "${{ github.action_path }}" >> $Env:GITHUB_PATH
+        (Resolve-Path "${{ github.action_path }}/../../../Scripts").Path >> $Env:GITHUB_PATH
+
+    - name: Add MSBuild to PATH
+      # v1.1.3
+      uses: microsoft/setup-msbuild@34cfbaee7f672c76950673338facd8a73f637506
+      with:
+        vs-version: "[17.3,]"
```

Comment:
Yes. The current latest version of VS is 17.4.1, so this is the one we know for sure current projects work. That's not available in the GHA image, however, and the next best thing is 17.3 (while 17.x would most possibly work fine, there's no advantage to allowing that, since the runner images won't downgrade).